Dynatrace/dynatrace-operator
# Update `k8s.io/{api,apiextensions-apiserver}` dependencies

# Description

Cherry-picked 67ff210ab625abccdc27c75a53f79bbc137d57f1 and e64f1ca52c8b77e43e69ff197e4d4e13616e4ab1 dependabot PRs.

Switch from `configmap` to `configmapleases` as lock object for leaderselection - as described: https://github.com/kubernetes-sigs/controller-runtime/pull/1773.

```
10:55:01.401660       1 leaderelection.go:248] attempting to acquire leader lease dynatrace/dynatrace-operator-lock...
10:55:01.405545       1 leaderelection.go:258] successfully acquired lease dynatrace/dynatrace-operator-lock          
```

## How can this be tested?
- CI
- run operator
- scale operator deployment to >1 and verify if leaderselection works correctly


## Checklist
~- [ ] Unit tests have been updated/added~
